src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -719,12 +779,18 @@ private byte[] GetAsciiEncoding(byte value)
             {
                 if (!WillEncode(value))
                 {
-                    _asciiEscape[value] = s_noEscape;
-                    return s_noEscape;
+                    encoding = Array.Empty<byte>();
+                    _asciiEscape[value] = encoding;
                 }
             }
 
             return encoding;
         }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
@AndyAyersMS - I recall you saying something in the past re: methods that throw exceptions shouldn't be marked _noinlining_, since this attribute could prevent the JIT from recognizing the call site as invoking a no-return method. Is this still the recommendation?

Reply:
Yes, it's still the case.

Reply:
I will remove the NoInlining here.

However when using the "throw helper" pattern which helps reduce the [size of code for generics methods](https://github.com/dotnet/coreclr/issues/22400) there are several cases (at least in System.Text.Json) where NoInlining is specified. I assume because otherwise it is possible the throwhelper method would just be inlined (?).

Reply:
Without the NoInlining, throw helpers will not get inlined.

The jit knows how to recognize these methods and will avoid inlining them and also mark the throw helper call as cold so that the call site is moved to the end of the method. This latter bit gives better hot code locality and usually eliminates a taken branch.

In principle you can use the jit-diff tool from [jitutils](https://github.com/dotnet/jitutils) to look at the impact on the jit-generated code with and without the attribute(s). This currently requires a checked jit so isn't as CoreFx friendly as one would like. I would expect to see similar size overall with and without the NoInlining attribute on throw helpers, but the code flow in calling methods will be less optimal with the attribute in place.